### DISCUSSION TITLE
elegant-code-base

### DISCUSSION THREAD

**Discussion by giscus** [sunface/rust-by-practice#203] [Book Comments]
# elegant-code-base

Learning Rust By Practice, narrowing the gap between beginner and skilled-dev with challenging examples, exercises and projects.

https://zh.practice.rs/elegant-code-base.html

**Comment by kosl90** [comment]
编辑器的项目可以考虑换成这个 [https://medium.com/@otukof/678a463f968b?source=friends_link&sk=e2e89383cf8abcf35a7d0afddf8fadc2](https://medium.com/@otukof/678a463f968b?source=friends_link&sk=e2e89383cf8abcf35a7d0afddf8fadc2)，或者加个注释。

现在链接的项目使用了 termion，这个 crate 在 windows 无法编译使用，medium 上的这个项目是使用的 crossterm，在 windows 上可以编译。

**Reply by sunface** [reply]
感谢建议。。如果可以的话，请个 PR 最好😆

**Reply by iscas-zac** [reply]
2023/3/7，文中的编辑器项目链接失效了，搜索引擎返回的网址是https://www.flenker.blog/hecto/

**Comment by guanguoliang** [comment]
有没有一些项目有构建过程的，就是有个引导从哪里开始，直接看一个项目对初学者来说有点摸不着北

**Comment by OnlyAIGC** [comment]
希望可以推荐一些im的练手小项目（笑）

**Comment by coder0xe** [comment]
真强啊
